packages/adp-tooling/src/client/target-systems.ts

```diff
@@ -0,0 +1,126 @@
+ *
+ * @param {BackendSystem} system - The backend system to transform.
+ * @returns {Endpoint} The transformed endpoint object.
+ */
+export const transformBackendSystem = (system: BackendSystem): Endpoint => ({
+    Name: system.name,
+    Url: system.url,
+    Client: system.client,
+    UserDisplayName: system.userDisplayName,
+    Scp: !!system.serviceKeys,
+    Authentication: system.authenticationType,
+    Credentials: {
+        username: system.username,
+        password: system.password
+    }
+});
+
+/**
+ * Service class to manage and retrieve information about system systems,
+ * including their names, authentication requirements, and specific details.
+ */
+export class TargetSystems {
+    private systems: Endpoint[];
+
+    /**
+     * Creates an instance of EndpointsManager.
+     *
+     * @param {ToolsLogger} logger - The logger.
+     */
+    constructor(private readonly logger: ToolsLogger) {}
+
+    /**
+     * Returns the stored systems.
+     *
+     * @returns {Promise<Endpoint[]>} An array of system systems.
+     */
+    public async getSystems(): Promise<Endpoint[]> {
+        if (!this.systems) {
+            this.systems = await this.loadSystems();
+        }
+        return this.systems;
+    }
+
+    /**
+     * Loads systems from either BAS (via destinations) or from the backend system store.
+     *
+     * @returns {Promise<Endpoint[]>} A promise that resolves with an array of endpoints.
+     * @throws {Error} If fetching the systems fails.
+     */
+    private async loadSystems(): Promise<Endpoint[]> {
+        try {
+            let endpoints: Endpoint[] = [];
+            if (isAppStudio()) {
+                const destinations = await listDestinations({
+                    stripS4HCApiHosts: true
+                });
+                endpoints = Object.values(destinations).filter((dest) => dest.WebIDEUsage?.includes('dev_abap'));
+            } else {
+                const systemStore = await getService<BackendSystem, BackendSystemKey>({
+                    entityName: 'system'
+                });
+                const backendSystems = await systemStore?.getAll();
+                endpoints = backendSystems.map(transformBackendSystem);
+            }
+            return endpoints;
+        } catch (e) {
+            this.logger?.error(`Failed to fetch systems list. Reason: ${e.message}`);
+            throw new Error(e.message);
+        }
+    }
+
+    /**
+     * Retrieves a particular system by its name.
+     *
+     * @param {string} name - The system name or URL.
+     * @returns {Promise<Endpoint | undefined>} System details including client, url, and credentials, or undefined if not found.
+     */
+    public async getSystemByName(name: string): Promise<Endpoint | undefined> {
+        const systems = await this.getSystems();
+        const system = systems.find((e) => e.Name === name || e.Url === name);
```

Comment:
Generally this is ok, because we have code implemented to prevent systems being saved with the same name. However, this could be done by manually changing the system name in the `systems.json` file for example. As the key for the system is a combination of the URL + Client, it is generally safer to search for systems using this combination.

Reply:
@cianmSAP We don't have any access to the URL or the client at this stage. I am not sure that we can do that right now, we might return to this discussion in the next PRs. Do you think there could be cases like that where people would go and change the settings file manually? I don't think we should handle such cases.

Reply:
Hmm if you have queried the store to fetch the names, you would have access to the URL/Client in that same call?
No I don't expect it to happen, hence why I said it's generally ok, but the name is not a strict UID. The URL+Client combo is a safer method to determine a certain system.

We can revisit it.